.github/workflows/release.yml

```diff
@@ -0,0 +1,94 @@
+name: Build and Release
+
+env:
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  
+on:  
+  push:
+    tags:
+      - '*'
+
+jobs:
+  build-and-upload:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.14.13' 
+      
+      - if: matrix.os == 'windows-latest'
+        name: Install make in windows
+        uses: crazy-max/ghaction-chocolatey@v1
+        with:
+          args: install make
+      
+      - name: Setup Go environment
+        run: make install
+      
+      - name: Read version.txt
+        id: version
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./version.txt
+      
+      - name: Build go-spacemesh
+        run: make build GO_SPACEMESH_VERSION=${{ steps.version.outputs.content }} WINDOWS_BUILD_DIR=./build
```

Comment:
Maybe add a comment here to indicate that `make build` reads `version.txt` by default, but you're passing in `VERSION` here to be explicit